[Test] Run copy-propagation on SILOptimizer/opaque_values_mandatory.sil.

nate-chandler opened the pull request:
<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!-- If this pull request resolves any bugs in the Swift bug tracker, provide a link: -->
Resolves SR-NNNN.

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/master/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

nate-chandler committed:
[Test] Run copy-propagation on SILOptimizer/opaque_values_mandatory.sil.

nate-chandler committed:
[SILOpt] Added -enable-copy-propagation flag.

The new flags sets the EnableCopyPropagation flag on the SILOpt[ion]s
class used to determine the behavior of sil-opt.

nate-chandler commented:
@swift-ci please test

atrick reviewed (approved):
This run-line looks good to me.

 -enable-ossa-modules currently enables non-mandatory copy propagation at -O only just so we can get a good performance baseline.

atrick commented on tools/sil-opt/SILOpt.cpp:
```diff
@@ -122,6 +122,10 @@ static llvm::cl::opt<bool> EnableOSSAModules(
                    "this is disabled we do not serialize in OSSA "
                    "form when optimizing."));
 
+static llvm::cl::opt<bool> EnableCopyPropagation(
```
We actually have two options -enable-copy-propagation and -disable-copy-propagation, both of which default to 'false'. Enabling either one overrides the default pipeline behavior, which depends on the optimization level and whether OSSA is enabled.

nate-chandler replied:
It looks like swift-frontend has those flags to control the `EnableCopyPropagation` and `DisableCopyPropagation` flags of `the `SILOptions` class sil-opt doesn't.  Considering that they both have independent effects, it seems likely that at some point they'll both be useful for `sil-opt`.  I went ahead and pushed another commit to add it.

The test failures were
```
  Swift(macosx-x86_64) :: Interpreter/SDK/objc_bridge_cast.swift
  Swift(macosx-x86_64) :: Interpreter/SDK/objc_dealloc.swift
```
and seem to be affecting unrelated, affecting other PRs, like https://ci.swift.org/job/swift-PR-macos/28081/consoleText .

nate-chandler committed:
[SILOpt] Added -disable-copy-propagation flag.

The new flags sets the DisableCopyPropagation flag on the SILOpt[ion]s
class used to determine the behavior of sil-opt.

nate-chandler commented:
@swift-ci please test

atrick reviewed (approved):
LGTM

nate-chandler commented:
@swift-ci please test windows platform

nate-chandler commented:
@swift-ci please test windows platform

nate-chandler commented:
@swift-ci please test windows platform

nate-chandler commented:
@swift-ci please clean test windows platform

nate-chandler commented:
@swift-ci please clean test windows platform
